plugins/coverbrowser.koplugin/mosaicmenu.lua

```diff
@@ -765,15 +775,32 @@ function MosaicMenuItem:paintTo(bb, x, y)
         end
         local iy = self.height - math.ceil((self.height - target.dimen.h)/2) - corner_mark:getSize().h
         -- math.ceil() makes it looks better than math.floor()
+        if self.status == "abandoned" then
+            corner_mark = abandoned_mark
+        elseif self.status == "complete" then
+            corner_mark = complete_mark
+        else
+            corner_mark = reading_mark
+        end
         corner_mark:paintTo(bb, x+ix, y+iy)
     end
 
-    if show_progress_in_mosaic and self.percent_finished then
+    if self.show_progress_bar then
         local cover_item = self[1][1][1]
-        local width = math.min(self.width * 0.60, cover_item.width)
-        progress_widget.width = width
-        local pos_x = x + self.width - progress_widget.width - math.ceil((self.width - cover_item.width) / 2) - progress_widget.bordersize
-        local pos_y = y + self.height - progress_widget.height - math.ceil((self.height - cover_item.height) / 2)
+        progress_widget.width = cover_item.width - Screen:scaleBySize(16)
+        local pos_x = x + math.floor((self.width - cover_item.width) / 2) + Screen:scaleBySize(8)
+        if self.do_hint_opened then
+            progress_widget.width = cover_item.width - corner_mark_size - Screen:scaleBySize(16)
```

Comment:
Just asking (dunno the answer): is it alright to use here `Screen:scaleBySize(16)` (probably some experimental value you found nice :), and not some fraction of `cover_item.width` or `corner_mark_size` ?
Screen:scaleBySize() will scale according to the screen size.
It may end up that all other things also end up being a fraction of the screen size, so we end up with all things sized related to the screen size, but I dunno :)

Reply:
It's probably related to the corner mark size, let me draw up some golden ratio charts and whatnot and figure out the fraction

Reply:
Done, it is indeed a derivative of `corner_mark_size`.